index.js

```diff
@@ -508,6 +508,42 @@ class Batch {
     return ite.next()
   }
 
+  createRangeIterator (opts = {}) {
+    const extension = (opts.extension === false && opts.limit !== 0) ? null : this.tree.extension
+
+    if (extension) {
+      const { onseq, onwait } = opts
+      let version = 0
+      let next = 0
+
+      opts = encRange(this.keyEncoding, {
+        ...opts,
+        sub: this.tree._sub,
+        onseq (seq) {
+          if (!version) version = seq + 1
+          if (next) next--
+          if (onseq) onseq(seq)
+        },
+        onwait (seq) {
+          if (!next) {
+            next = Extension.BATCH_SIZE
+            extension.iterator(ite.snapshot(version))
+          }
+          if (onwait) onwait(seq)
+        }
+      })
+    } else {
+      opts = encRange(this.keyEncoding, { ...opts, sub: this.tree._sub })
+    }
+
+    const ite = new RangeIterator(new Batch(this.tree, this.feed.snapshot(), null, false, opts), opts)
```

Comment:
This `new Batch(...)` seems a bit off, not sure what to do there

Reply:
Because is creating a new batch inside of an already batch

Reply:
Passing `this` (self batch) works but don't know if that's correct, because original `createRangeIterator` was using `snapshot()`

For example, original code:
```js
const ite = new RangeIterator(new Batch(this.tree, this.feed.snapshot(), null, false, opts), opts)
```

vs

```js
this = new Batch(this, this.feed, mutexify(), true, opts)
// ...
const ite = new RangeIterator(this, opts)
```